modules/opensearch/opensearch.client.inc

```diff
@@ -660,17 +660,29 @@ function _opensearch_cache(TingClientRequest $request, $response = NULL) {
   // and we want to retrieve the cache value and where $value is set to NULL
   // explicitly.
   if (func_num_args() == 2) {
-    // Update the cache with the provided value.
-    $static_cache[$cid] = $response;
+    // Update the cache with the provided value. If you here just assign the
+    // response to the static variable here even though it's deep clone in the
+    // return statement changes to the collections inside the object from
+    // outside this function changes the cached object.
+    $static_cache[$cid] = unserialize(serialize($response));
```

Comment:
This is a repeating pattern.

I suggest implementing a `_opensearch_deep_clone($object)` function where the DocBlock can describe the reasoning behind this.